tests/js/counter.spec.js

```diff
@@ -0,0 +1,148 @@
+    await typeInInputWithLabel( 'Target', '100' );
+    await typeInTextareaWithLabel( 'Text', 'Signature  collected %completed% out of %target%, the remaining is %remaining%' );
+
+    expect( await getEditedPostContent() ).toMatchSnapshot();
+  }, 50000 );
+
+  it( 'test the counter block with Progress Bar', async () => {
+    await selectBlockByName( 'planet4-blocks/counter' );
+
+    // Add richtext inputs.
+    await typeInInputWithPlaceholderLabel( 'Enter title', 'Progress Bar counter block' );
+    await typeInInputWithPlaceholderLabel( 'Enter description', 'Counter block description' );
+
+    // Select the Progress Bar counter style
+    await openSidebarPanelWithTitle( 'Styles' );
+    await selectStyleByName( 'Progress Bar' );
+
+    // Add Setting inputs
+    await openSidebarPanelWithTitle( 'Setting' );
+    await typeInInputWithLabel( 'Completed', '75' );
+    await typeInInputWithLabel( 'Target', '100' );
+    await typeInTextareaWithLabel( 'Text', 'Signature  collected %completed% out of %target%, the remaining is %remaining%' );
+
+    // The Progress Bar component should appear.
+    await expect( page ).toMatchElement( '.progress-bar' );
+
+    expect( await getEditedPostContent() ).toMatchSnapshot();
+  }, 50000 );
+
+  it( 'test the counter block with Progress Dial', async () => {
+    await selectBlockByName( 'planet4-blocks/counter' );
+
+    // Add richtext inputs.
+    await typeInInputWithPlaceholderLabel( 'Enter title', 'Progress Dial counter block' );
+    await typeInInputWithPlaceholderLabel( 'Enter description', 'Counter block description' );
+
+    // Select the Progress Dial counter style
+    await openSidebarPanelWithTitle( 'Styles' );
+    await selectStyleByName( 'Progress Dial' );
+
+    // Add Setting inputs
+    await openSidebarPanelWithTitle( 'Setting' );
+    await typeInInputWithLabel( 'Completed', '75' );
+    await typeInInputWithLabel( 'Target', '100' );
+    await typeInTextareaWithLabel( 'Text', 'Signature  collected %completed% out of %target%, the remaining is %remaining%' );
+
+    // The Progress Dial component should appear.
+    await expect( page ).toMatchElement( '.progress-arc' );
+
+    expect( await getEditedPostContent() ).toMatchSnapshot();
+  }, 50000 );
+
+  it( 'test the counter block with API input', async () => {
+    await selectBlockByName( 'planet4-blocks/counter' );
+
+    // Add richtext inputs.
+    await typeInInputWithPlaceholderLabel( 'Enter title', 'Progress Dial counter block' );
+    await typeInInputWithPlaceholderLabel( 'Enter description', 'Counter block description' );
+
+    // Select the Text only counter style
+    await openSidebarPanelWithTitle( 'Styles' );
+    await selectStyleByName( 'Progress Dial' );
+
+    await setUpResponseMocking( COUNTER_API_MOCK_RESPONSES );
+
+    // Add Setting inputs
+    await openSidebarPanelWithTitle( 'Setting' );
+    // TO DO: Need some changes in counter block to work below line of code.
```

Comment:
What is the issue here?

Reply:
The api url is a long string and on each keypress the react preview got refresh in test execution, which turns into test timeout. I am thinking to use similar solution like [spreadsheet block](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/master/assets/src/blocks/Spreadsheet/SpreadsheetEditor.js#L92)

Reply:
Is there a ticket for this issue?

Reply:
Nope I just added it ( [PLANET-5372](https://jira.greenpeace.org/browse/PLANET-5372) ) (I was tring to fix it my own but fall into many other errors)
Would you like to take it forward?

Reply:
The issue was solved by [adding a catch](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/331/commits/67f3890b8c6af97c54a27e9bea2b323da17223b8) on fetch() function.